connectors/sources/google_drive.py

```diff
@@ -284,88 +565,137 @@ def _validate_service_account_json(self):
+            dict: Access control doc.
         """
-
-        async for drive in self._google_drive_client.api_call_paged(
-            resource="drives",
-            method="list",
-            fields="nextPageToken,drives(id,name)",
-            pageSize=100,
+        user_id = user.get("id")
+        user_email = user.get("primaryEmail")
+        user_domain = _get_domain_from_email(user_email)
+        user_groups = []
+        async for groups_page in self.google_admin_directory_client.list_groups_for_user(
+            user_id
         ):
-            yield drive
+            for group in groups_page.get("groups", []):
+                user_groups.append(group.get("email"))
 
-    async def retrieve_all_drives(self):
-        """Retrieves all shared drives from Google Drive
+        user_access_control = [
+            _prefix_user(user_email),
+            _prefix_domain(user_domain),
+        ] + [_prefix_group(group) for group in user_groups]
 
-        Returns:
-            dict: mapping between drive id and its name
-        """
-        drives = {}
-        async for page in self.get_drives():
-            drives_chunk = page.get("drives", [])
-            for drive in drives_chunk:
-                drives[drive["id"]] = drive["name"]
+        return {
+            "_id": user_email,
+            "identity": {
+                "name": user.get("name").get("fullName"),
+                "email": user_email,
+            },
+        } | self.access_control_query(access_control=user_access_control)
 
-        return drives
+    async def prepare_access_control_documents(self, users_page):
+        """Generate access control document.
 
-    async def get_folders(self):
-        """Fetch all folders (id, name, parent) from Google Drive
+        Args:
+            users_page (list): List with user objects.
 
         Yields:
-            dict: Folder metadata.
+            dict: Access control doc.
         """
-        async for folder in self._google_drive_client.api_call_paged(
-            resource="files",
-            method="list",
-            corpora="allDrives",
-            fields="nextPageToken,files(id,name,parents)",
-            q=f"mimeType='{FOLDER_MIME_TYPE}' and trashed=false",
-            includeItemsFromAllDrives=True,
-            supportsAllDrives=True,
-            pageSize=1000,
-        ):
-            yield folder
+        users = users_page.get("users", [])
 
-    async def retrieve_all_folders(self):
-        """Retrieves all folders from Google Drive
+        async def process_user(user, semaphore):
+            async with semaphore:
+                return await self.prepare_single_access_control_document(user=user)
 
-        Returns:
-            dict: mapping between folder id and its (name, parents)
+        # Create the shared semaphore, it controls how many concurrent
+        # groups/list requests can be open at any given time
+        semaphore = asyncio.Semaphore(20)
```

Comment:
For [other connectors](https://github.com/search?q=repo%3Aelastic%2Fconnectors-python%20MAX_CONCURRENCY%20%3D%20&type=code) I see it's defined as a `MAX_CONCURRENCY` and hardcoded as a constant. 

Since we don't expose `retry_count` I'm not sure if we want to expose this concurrency setting to the end user. Instead we can choose a reasonably high value and use it as a constant.

Reply:
Maybe we want to double check with product here, if that should be an advanced configuration setting or not. I think we want to make things as configurable as possible. cc @danajuratoni

Reply:
Dana suggested to put this option as an advanced RCF - will follow up adding this field

Reply:
Current approach is to keep the connector configuration in Kibana as simple as possible, and not encourage users to play with deployment options in Elastic Cloud. Advanced RCF would still be configurable for connector clients, even if they don't show up in Kibana and is aligned with the current shift in mindset, to be more dev focused. 
++ on using similar features consistently(such as a MAX_CONCURRENCY param) across all connectors

We might want to discuss in the future options to show advanced RFCs for connector clients in Kibana, even if they're hidden for native connectors, but that is not in the 8.10 scope.